src/util/bundleUtils.ts

```diff
@@ -14,3 +20,90 @@ export function createSearchsetBundle<T extends fhir4.FhirResource>(entries: T[]
     entry: entries.map(e => ({ resource: e }))
   };
 }
+
+/**
+ * Takes in a measure resource, finds all dependent library resources and bundles them
+ * together with the measure in a collection bundle
+ */
+export async function createMeasurePackageBundle(measure: fhir4.Measure): Promise<fhir4.Bundle<fhir4.FhirResource>> {
+  logger.info(`Assembling collection bundle from Measure ${measure.id}`);
+  if (measure.library && measure.library.length > 0) {
+    const [mainLibraryRef] = measure.library;
+    const mainLibQuery = getQueryFromReference(mainLibraryRef);
+    const libs = await findResourcesWithQuery(mainLibQuery, 'Library');
+    if (!libs || libs.length < 1) {
+      throw new ResourceNotFoundError(`Could not find Library ${mainLibraryRef} referenced by Measure ${measure.id}`);
+    }
+    const mainLib = libs[0];
+
+    const allLibsDups = await getAllDependentLibraries(mainLib as fhir4.Library);
+    // de-dup by id using map
+    const idMap = new Map(allLibsDups.map(lib => [lib.id, lib]));
+    const allLibs = Array.from(idMap.values());
+    const result: fhir4.Bundle = { resourceType: 'Bundle', type: 'collection', id: v4() };
+    result.entry = [{ resource: measure }];
+    result.entry.push(
+      ...allLibs.map(r => ({
+        resource: r
+      }))
+    );
+    return result;
+  } else {
+    throw new BadRequestError(`Uploaded measure: ${measure.id} does not reference a Library`);
+  }
+}
+
+/**
+ * Assemble a mongo query based on a reference to another resource
+ * @param {string} reference assumed to be canonical
+ * @returns {Filter} mongo query to pass in to mongo controller to search for the referenced resource
```

Comment:
We don't necessarily need these since we are in TypeScript, right? Although I do think it's good to have that the string param is assumed to be canonical so we can either keep it or just include it in the overall description.

Reply:
Yeah I've come around to the take of basically it's okay when some stricter assumption needs to be clarified about something.

Like if it was just `@param {string} reference the reference` I'd say delete, but this is probably fine